### PR TITLE
BRAIN-15566 - Access model state with get_state() function

### PR DESCRIPTION
**Pull request by adarmiento**
### Changes

* Implemented get_state() function in BaseMab abstract class

### CODE REVIEW

File: tests/test_cmab.py [hunk 1]
```diff
@@ -311,6 +316,33 @@ def run_predict(mab):
     assert mab != create_cmab_bernoulli_cold_start(action_ids=["a1", "a2", "a3", "a4", "a5"], n_features=n_features)
     run_predict(mab=mab)
 
+    @pytest.mark.parametrize("action_dict, action_ids", [
```
Comment (dariodandrea) [inline tests/test_cmab.py:319]: You can use hypothesis as in all the other tests. It allows you to try many more combinations with much less and clean code.

File: pybandits/base.py [hunk 2]
```diff
@@ -187,3 +187,17 @@ def predict(self, forbidden_actions: Optional[Set[ActionId]] = None):
...
+           Access the complete model internal state, enough to create an exact copy of the same model from it.
+           Returns
+           -------
+           model_class_name: str
+               The name of the class of the model.
+           model_state: dict
+               The internal state of the model (actions, scores, etc.).
+           """
+        model_name = self.__class__.__name__
+        state: dict = self.dict()
```
Comment (dariodandrea) [inline pybandits/base.py:202]: Strategy is empty if you use self.dict() (look at the tests). Do you see it as a problem?

Comment (adarmiento) [reply pybandits/base.py:202]: No, it's not a problem, as this represents the state, and some strategies are stateless.

File: pybandits/base.py [hunk 3]
```diff
@@ -129,9 +129,9 @@ def _get_valid_actions(self, forbidden_actions: Optional[Set[ActionId]]) -> Set[
         return valid_actions
 
     def _check_update_params(
-        self,
-        actions: List[ActionId],
-        rewards: List[Union[NonNegativeInt, List[NonNegativeInt]]],
+            self,
```
Comment (dariodandrea) [inline pybandits/base.py:132]: I guess this indentation is a mistake, right?

File: pybandits/base.py [hunk 4]
```diff
@@ -152,11 +152,11 @@ def _check_update_params(
     @abstractmethod
     @validate_arguments
     def update(
-        self,
-        actions: List[ActionId],
-        rewards: List[Union[BinaryReward, List[BinaryReward]]],
-        *args,
-        **kwargs,
+            self,
```
Comment (dariodandrea) [inline pybandits/base.py:155]: See avobe

File: pybandits/base.py [hunk 5]
```diff
@@ -187,3 +187,17 @@ def predict(self, forbidden_actions: Optional[Set[ActionId]] = None):
         probs: List[Dict[ActionId, float]] of shape (n_samples,)
             The probabilities of getting a positive reward for each action.
         """
+
+    def get_state(self) -> (str, dict):
+        """
+           Access the complete model internal state, enough to create an exact copy of the same model from it.
```
Comment (dariodandrea) [inline pybandits/base.py:193]: Can you indent all the docstring at the same way?

File: pybandits/base.py (hunk 3 above)
Comment (adarmiento) [reply pybandits/base.py:132]: I guess it's some auto-format thing. 
Reverting

**Review by dariodandrea — APPROVED** [review]
Good job @adarmiento! Let's merge :)

**Pull request merged**